File: app/src/main/java/pl/butajlo/androidadvanced/ui/ScreenNavigator.java
package pl.butajlo.androidadvanced.ui;

import com.bluelinelabs.conductor.Controller;
import com.bluelinelabs.conductor.Router;

public interface ScreenNavigator {

    void initWithRouter(Router router, Controller rootScreen);

    /**
     * Pop the actual screen
     * @return true if the screen was popped, false if not
     */
    boolean pop();

    /**
     * It is used to clean the router, if the activity is destroyed
     */
    void clear();
}

File: app/src/main/java/pl/butajlo/androidadvanced/ui/ScreenNavigator.kt
package pl.butajlo.androidadvanced.ui

import com.bluelinelabs.conductor.Controller
import com.bluelinelabs.conductor.Router

interface ScreenNavigator {

    /**
     * Pop the actual screen
     * @return true if the screen was popped, false if not
     */
    fun pop(): Boolean

    fun goToRepoDetails(repoOwner: String, repoName: String)
}